codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ConfigGenerator.java

```diff
@@ -100,7 +100,7 @@ private void renderConfigDocumentation(RubyCodeWriter writer) {
         writer.writeYardMethod("initialize(*options)", () -> {
             clientConfigList.forEach((clientConfig) -> {
                 String member = RubyFormatter.asSymbol(RubySymbolProvider.toMemberName(clientConfig.getName()));
-                String returnType = clientConfig.getType();
+                String returnType = clientConfig.getDocumentationType();
```

Comment:
It'd be nice if we can keep using type here but special case Boolean instead of using documentation type, because we may want to have a responds to documentation in the future.

Reply:
Initially, I was looking for areas where we used `getType()` method to see how it was used throughout the codegen. Then, came across this piece. I thought updating to  using `getDocumentationType()` was appropriate since the method, `renderConfigDocumentation`  is to render documentation. 

> because we may want to have a responds to documentation in the future

Could you expand on this? I don't follow (might be due to lack of context)

Reply:
While I did remove `type` from ClientConfig constructor, the `type` method still exists on the Builder for convenience and to avoid breaking existing code (also creates type constraint and takes care of `documentationType` if unset). You could still skip using the `type` builder method to just set the `documentationType` like below:
```
ClientConfig.builder()
  .constraint(new RespondsToConstraint(":call"))
  .documentationType("Callable")
  .build()
```
I believe this will address the above.